src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/CustomCATest.java

```diff
@@ -185,8 +203,36 @@ private void subscribeToCertificateUpdates(GetCertificateRequest request)
         api.subscribeToCertificateUpdates(request);
     }
 
+    private void waitForCertificateAuthorityToBe(X509Certificate expectedCA) throws InterruptedException {
+        CertificateStore store = kernel.getContext().get(CertificateStore.class);
+
+        TestHelpers.eventuallyTrue(() -> {
+            try {
+                String expectedPem = CertificateHelper.toPem(expectedCA);
+                String configuredCA = CertificateHelper.toPem(store.getCACertificate());
+               return expectedPem.equals(configuredCA);
+            } catch (CertificateException | KeyStoreException | IOException e) {
+                return  false;
+            }
+        });
+    }
+
+    private void waitForCertificateAuthorityToChange() throws KeyStoreException, InterruptedException {
+        CertificateStore store = kernel.getContext().get(CertificateStore.class);
+        X509Certificate currentCA = store.getCACertificate();
+        TestHelpers.eventuallyTrue(() -> {
+            try {
+                String pem = CertificateHelper.toPem(store.getCACertificate());
+                String currentCAPem = CertificateHelper.toPem(currentCA);
+                return !pem.equals(currentCAPem);
+            } catch (KeyStoreException | CertificateEncodingException | IOException e) {
+                return false;
+            }
+        });
+    }
```

Comment:
Race-y. If the CA changes before this method is called then it will fail. Previous CA needs to be retrieved before the action and then passed in

Reply:
Yeah it won't cause a race right now because of where I am using it but you are right has the potential for race 🐎